test/runtests.jl

```diff
@@ -78,6 +78,15 @@ end
         IntegralArray(X, X)
         @test iX == X == IntegralArray(reshape(1:25, 5, 5))
     end
+    
+    @testset "small types" begin
+        X = Int8.(collect(reshape(1:25, 5, 5)))
+        iX = similar(X)
+        iAa = IntegralArray(X)
+        iAb = IntegralArray(iX, X)
```

Comment:
and thus this should still error

```suggestion
        @test_throws InexactError IntegralArray(iX, X)
```